python/sqlflow_submitter/xgboost/predict.py

```diff
@@ -32,40 +32,52 @@ def pred(datasource,
          hdfs_pass="",
          pai_table="",
          model_params=None,
-         train_params=None):
+         train_params=None,
+         transform_fn=None,
+         feature_column_code=""):
     if not is_pai:
         conn = db.connect_with_data_source(datasource)
     else:
         conn = None
     label_name = label_meta["feature_name"]
-    dpred = xgb_dataset(datasource,
-                        'predict.txt',
-                        select,
-                        feature_metas,
-                        feature_column_names,
-                        None,
-                        is_pai,
-                        pai_table,
-                        True,
-                        True,
-                        batch_size=DEFAULT_PREDICT_BATCH_SIZE
-                        )  # NOTE: default to use external memory
+    dpred = xgb_dataset(
+        datasource=datasource,
+        fn='predict.txt',
+        dataset_sql=select,
+        feature_specs=feature_metas,
+        feature_column_names=feature_column_names,
+        label_spec=None,
+        is_pai=is_pai,
+        pai_table=pai_table,
+        pai_single_file=True,
+        cache=True,
+        batch_size=DEFAULT_PREDICT_BATCH_SIZE,
+        transform_fn=transform_fn,
+        feature_column_code=feature_column_code,
+        raw_data_dir="predict.raw.dir")  # NOTE: default to use external memory
     bst = xgb.Booster({'nthread': 4})  # init model
     bst.load_model("my_model")  # load data
     print("Start predicting XGBoost model...")
+
+    if is_pai:
+        pai_table = "odps://{}/tables/{}".format(*pai_table.split("."))
```

Comment:
What if pai_table doesn't contains maxcompute project name?
Will we append the project name before code_gen?

Reply:
We have appended project name [here](https://github.com/sql-machine-learning/sqlflow/blob/develop/pkg/sql/pai_submitter.go#L105).